kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DefaultImporter.java

```diff
@@ -115,13 +115,20 @@ private void write() {
     }
     
     private PipelineJobProgressUpdatedParameter flush(final DataSource dataSource, final List<Record> buffer) {
-        List<GroupedDataRecord> result = MERGER.group(buffer.stream().filter(each -> each instanceof DataRecord).map(each -> (DataRecord) each).collect(Collectors.toList()));
+        List<DataRecord> dataRecords = buffer.stream().filter(each -> each instanceof DataRecord).map(each -> (DataRecord) each).collect(Collectors.toList());
         int insertRecordNumber = 0;
         int deleteRecordNumber = 0;
+        for (DataRecord each : dataRecords) {
+            if (IngestDataChangeType.INSERT.equals(each.getType())) {
+                insertRecordNumber++;
+            }
+            if (IngestDataChangeType.DELETE.equals(each.getType())) {
+                deleteRecordNumber++;
+            }
```

Comment:
Currently, `processedRecordsCount = insertRecordNumber - deleteRecordNumber`.
Looks `deleteRecordNumber` should be ignored, since the deleted records must be inserted before on migration case, so it's already processed.